### PR TITLE
[NativeAOT] Make conservative stack reporting configurable.

### PR DESCRIPTION
**Pull request by VSadov**
Conservative stack reporting is a convenient option when investigating issues with suspension/unwinding/root reporting.
Having to rebuild the runtime and testcases in order to switch is not convenient though as switching back and forth takes a very long time.

This change allows turning conservative stack reporting off/on via environment variable.

### CODE REVIEW

File: src/coreclr/nativeaot/Runtime/RhConfigValues.h [hunk 1]
```diff
@@ -16,6 +16,7 @@ DEBUG_CONFIG_VALUE_WITH_DEFAULT(BreakOnAssert, 1)
 RETAIL_CONFIG_VALUE(StressLogLevel)
 RETAIL_CONFIG_VALUE(TotalStressLogSize)
 RETAIL_CONFIG_VALUE(gcServer)
+RETAIL_CONFIG_VALUE(GcConservativeStacks)   // Enables conservative stack reporting
```
Comment (jkotas) [inline src/coreclr/nativeaot/Runtime/RhConfigValues.h:19]: We should call it the same thing as in CoreCLR (https://github.com/dotnet/runtime/blob/242c95aab1085dc5af9a017c00fe78c58386229f/src/coreclr/inc/clrconfigvalues.h#L268)

Comment (VSadov) [reply src/coreclr/nativeaot/Runtime/RhConfigValues.h:19]: `gcConservative` turns support for conservative reporting in the GC. Native AOT enables that unconditionally.
I think if we use the same key we will see it always "on".

Comment (jkotas) [reply src/coreclr/nativeaot/Runtime/RhConfigValues.h:19]: `gcConservative` turns on conservative stack scanning in CoreCLR: https://github.com/dotnet/runtime/blob/4cf2e3bb18162d750e7a128bb805c49c9fd52612/src/coreclr/vm/gcenv.ee.cpp#L129. 

> NativeAOT enables that unconditionally. I think if we use the same key we will see it always "on".

The GC should see it as always on in NativeAOT, but the runtime should use the env variable that you are adding.

Comment (VSadov) [reply src/coreclr/nativeaot/Runtime/RhConfigValues.h:19]: Actually it looks like we intercept `gcConservative` when GC asks about it. We may be able to use the same key.

Another issue could be that `gcConservative` would affect the test infrastructure as well as NativeAOT tests. That is probably ok.

**Review by jkotas — APPROVED** [review]

**Review by filipnavara — APPROVED** [review]

**Comment by VSadov** [comment]
Thanks!

**Pull request merged**